adr/0006-dev-discussions.adoc

```diff
@@ -0,0 +1,125 @@
+= Setup of Development Discussions  
+
+* Status: proposed
+* Date:  2024-07-18
+
+== Context and Problem Statement
+
+Currently development discussions happens in a variety of places, including the mailing list, zulip, github issues, pull-requests, discussions, calls etc.
+
+Its awesome that we have so many ways to communicate, but it also makes it difficult to find the information you need or being able to at least follow certain areas.
+ 
+This ADR proposes to have a more structured way of having development discussions, so that it is easier to find and follow such discussions.
+
+== Decision Drivers
+
+* Enable better open governance
+* Prefer async over sync communication
+* Avoid requiring to subscribe to a "raging river" of notifications that is not filterable
+* Allow use of e-mail as not everyone can/will use web-ui
+* Have consistent place to publish and lookup Quarkus development ideas/suggestions
+
+== Scenarios 
+
+For a Quarkus core developer, where do they best post ideas for suggestion and feedback? Zulip chat? Open an issue? Submit a PR directly? Post to quarkus-dev? Open a GitHub discussion? Use private chat? Today all of the above happens based on habit or specific requests.
+
+A dev interested in doing contributions by following the ongoing conversations - which of the many channels do they follow? 
+
+== Considered options 
+
+Primarily use quarkus-dev mailing list::
+    Advantages::::
+        * well-known and understood by most if not all existing Quarkus contributors (~1000+ subscribers)
+        * can post, comment and read with any e-mail client
+        * filters out most noise as people need to subscribe
+    Disadvantages::::
+        * over time it has died out, rarely where conversations starts/continues being discussed
+        * e-mail/google groups is just not where new people show up as much now
+        * requires subscribing to specific list to post
+        * spam (fixed now by moderation)
+        * arbitrary blocking of mails (specifally @gsmet and others)
```

Comment:
We should also mention that there's a steady stream of user questions on the quarkus-dev mailing list, and every time it happens one of us has to reply to point out it's not a mailing list for users.

Reply:
we should also be prepared, GH Discussion spam happens, too.

Reply:
I would not call it a "steady stream". It's been reduced greatly ever since we more consistently redirect. But yes it happens. 

Advantage of GitHub discussions here is we can transfer issues to the right place without the poster having to resend. 

I'll add that to list.

Reply:
That's a nice advantage, agreed.